### PR TITLE
Swift 4 Migration

### PR DESCRIPTION
**Pull request by gcutrini**
Updated code and dependencies.

Still missing implementing Codable API for JSON representation.

### CODE REVIEW

**Comment by colemancda** [comment]
I already implemented the Codable API. The code is hosted here: https://github.com/colemancda/CoreDataCodable/blob/master/Tests/CoreDataCodableTests/TestSummitJSON.swift

Basically there are 3 model types now:
`SummitResponse.Summit` -> `Model.Summit` -> `SummitManagedObject`

This is because the JSON can have a 1-1 representation to a decodable model, but there are values (e.g. `Summit.TimeZone`) that cannot be stored in CoreData as a attribute or as a relationship.

If you want you can copy and paste these 3 files to have the Codable model.

https://github.com/colemancda/CoreDataCodable/blob/master/Tests/CoreDataCodableTests/TestSummitJSON.swift
https://github.com/colemancda/CoreDataCodable/blob/master/Tests/CoreDataCodableTests/TestSummitManagedObject.swift
https://github.com/colemancda/CoreDataCodable/blob/master/Tests/CoreDataCodableTests/TestSummitModel.swift


And don't forget to add `[github "ColemanCDA/CoreDataCodable" "master"](https://github.com/colemancda/CoreDataCodable)` to your Cartfile.

**Comment by colemancda** [comment]
I also modified the CoreData Model `.xcmodel` so please copy that as well. It also wouldn't hurt to copy the unit tests. (TDD FTW)

File: OpenStack Summit/OpenStackSummit macOS/EventDetailViewController.swift [hunk 1]
```diff
@@ -376,3 +376,9 @@ final class EventDetailViewController: NSViewController, ContentController, Mess
         }
     }
 }
+
+extension NSStoryboardSegue.Identifier {
```
Comment (colemancda) [inline OpenStack Summit/OpenStackSummit macOS/EventDetailViewController.swift]: Nice!

**Comment by gcutrini** [comment]
@colemancda merging as is.
Implementing Codable model with CoreDataCodable compliance on another branch.
Creating Codable supporting types with JSON mapping and model jsonDecodable methods on yet another branch.

**Pull request merged**